migration/iD12/oracle/202406181038_IDEMPIERE-6177.sql

```diff
@@ -0,0 +1,78 @@
+;
+
+-- Jun 18, 2024, 10:40:23 AM CEST
+INSERT INTO AD_Element (AD_Element_ID,AD_Client_ID,AD_Org_ID,IsActive,Created,CreatedBy,Updated,UpdatedBy,ColumnName,Name,Description,Help,PrintName,EntityType,AD_Element_UU) VALUES (203937,0,0,'Y',TO_TIMESTAMP('2024-06-18 10:40:22','YYYY-MM-DD HH24:MI:SS'),100,TO_TIMESTAMP('2024-06-18 10:40:22','YYYY-MM-DD HH24:MI:SS'),100,'IsAutoGenerateInvoice','Generate Invoice','Automatically Generate Invoice after (Sales/Purchase) Order completed','When selected the Order will generate Invoice automatically. Note that only Orders with document subtype Standard and Prepay Order can be applied!','Generate Invoice','D','2c094bef-6468-4dea-9daf-c9b87795105e')
+;
+
+-- Jun 18, 2024, 10:42:18 AM CEST
+INSERT INTO AD_Column (AD_Column_ID,Version,Name,Description,Help,AD_Table_ID,ColumnName,DefaultValue,FieldLength,IsKey,IsParent,IsMandatory,IsTranslated,IsIdentifier,SeqNo,IsEncrypted,AD_Reference_ID,AD_Client_ID,AD_Org_ID,IsActive,Created,CreatedBy,Updated,UpdatedBy,AD_Element_ID,IsUpdateable,IsSelectionColumn,EntityType,IsSyncDatabase,IsAlwaysUpdateable,IsAutocomplete,IsAllowLogging,AD_Column_UU,IsAllowCopy,SeqNoSelection,IsToolbarButton,IsSecure,IsHtml,IsPartitionKey) VALUES (216612,0,'Generate Invoice','Automatically Generate Invoice after (Sales/Purchase) Order completed','When selected the Order will generate Invoice automatically. Note that only Orders with document subtype Standard and Prepay Order can be applied!',217,'IsAutoGenerateInvoice','N',1,'N','N','Y','N','N',0,'N',20,0,0,'Y',TO_TIMESTAMP('2024-06-18 10:42:18','YYYY-MM-DD HH24:MI:SS'),100,TO_TIMESTAMP('2024-06-18 10:42:18','YYYY-MM-DD HH24:MI:SS'),100,203937,'Y','N','D','N','N','N','Y','f1049fb6-356b-4768-be8a-5a2928bcb405','Y',0,'N','N','N','N')
+;
+
+-- Jun 18, 2024, 10:43:10 AM CEST
+INSERT INTO AD_Column (AD_Column_ID,Version,Name,Description,Help,AD_Table_ID,ColumnName,DefaultValue,FieldLength,IsKey,IsParent,IsMandatory,IsTranslated,IsIdentifier,SeqNo,IsEncrypted,AD_Reference_ID,AD_Client_ID,AD_Org_ID,IsActive,Created,CreatedBy,Updated,UpdatedBy,AD_Element_ID,IsUpdateable,IsSelectionColumn,EntityType,IsSyncDatabase,IsAlwaysUpdateable,IsAutocomplete,IsAllowLogging,AD_Column_UU,IsAllowCopy,SeqNoSelection,IsToolbarButton,IsSecure,IsHtml,IsPartitionKey) VALUES (216613,0,'Generate Shipment','Automatically Generate Shipment after (Sales/Purchase) Order completed','When selected the Order will generate Shipment automatically. Note that only Orders with document subtype Standard and Prepay Order can be applied!',217,'IsAutoGenerateInout','N',1,'N','N','Y','N','N',0,'N',20,0,0,'Y',TO_TIMESTAMP('2024-06-18 10:43:10','YYYY-MM-DD HH24:MI:SS'),100,TO_TIMESTAMP('2024-06-18 10:43:10','YYYY-MM-DD HH24:MI:SS'),100,203936,'Y','N','D','N','N','N','Y','dfb1cf89-a1a2-49c1-8389-0c60502e831c','Y',0,'N','N','N','N')
+;
+
+-- Jun 18, 2024, 10:43:12 AM CEST
+ALTER TABLE C_DocType ADD IsAutoGenerateInout CHAR(1) DEFAULT 'N' CHECK (IsAutoGenerateInout IN ('Y','N')) NOT NULL
```

Comment:
This is setting the flags to N for all the existing documents breaking backward compatibility.

You need to add an UPDATE to set the flags for existing documents:
shipment = Y for OnCredit, Warehouse, POS, Prepay
invoice = Y for OnCredit, POS, Prepay

Reply:
This should be nullable if it is not use for all doctype combination.

Reply:
See my comment on the ticket [here](https://idempiere.atlassian.net/browse/IDEMPIERE-6177?focusedCommentId=50929)